dashboard/app/models/user_module_task_assignment.rb

```diff
@@ -0,0 +1,34 @@
+# == Schema Information
+#
+# Table name: user_module_task_assignments
+#
+#  id                                   :integer          not null, primary key
+#  user_enrollment_module_assignment_id :integer
+#  professional_learning_task_id        :integer
+#  status                               :string(255)
+#
+# Indexes
+#
+#  task_assignment_to_module_assignment_index  (user_enrollment_module_assignment_id)
+#  task_assignment_to_task_index               (professional_learning_task_id)
+#
+
+# Maps a given user's module assignment to the tasks that they need to complete. More details are available on
+# http://wiki.code.org/display/Operations/Explanation+of+plc+Model
+
+class UserModuleTaskAssignment < ActiveRecord::Base
+  belongs_to :user_enrollment_module_assignment
+  belongs_to :professional_learning_task
+
+  def complete_assignment
+    self.status = :completed
+    self.save!
+
+    #Upon saving an assignment, there's a possibility that a course has been completed
+    user_course_enrollment = self.user_enrollment_module_assignment.user_professional_learning_course_enrollment
+
+    if (user_course_enrollment.user_module_task_assignment.where.not(status: :completed).count == 0)
```

Comment:
exists? is more efficient and a bit shorter than .count == 0 because it can stop as soon as it finds a result.

For readability, I think it's fine to call your temp variable just "enrollment" since it is used immediately and the context makes it unambiguous. 

With suggested changes, this would read:
````
if !enrollment.user_module_task_assignment.exists?(['status != ?', 'completed'])
````